webapp/src/utils/date/dateUtils.ts

```diff
@@ -0,0 +1,217 @@
+    if (lowerStr.startsWith(weekdayNames.abbreviation[i])) {
+      // Convert from 0-based Sunday=0 to 1-based Monday=1
+      return i === 0 ? 7 : i;
+    }
+  }
+
+  // Check short names
+  for (let i = 0; i < weekdayNames.short.length; i++) {
+    if (lowerStr.startsWith(weekdayNames.short[i].toLowerCase())) {
+      return i === 0 ? 7 : i;
+    }
+  }
+
+  return null;
+}
+
+/**
+ * Find month index from a string (1-based)
+ *
+ * @param dateStr - The string to search in
+ * @param locale - The locale to use
+ * @returns Month index (1-12) or null if not found
+ */
+export function findMonthIndex(dateStr: string, locale: SupportedLocale = "en"): number | null {
+  const lowerStr = dateStr.toLowerCase();
+  const monthNames = MONTH_NAMES[locale];
+
+  // Check abbreviations
+  for (let i = 0; i < monthNames.abbreviation.length; i++) {
+    if (lowerStr.includes(monthNames.abbreviation[i])) {
+      return i + 1;
+    }
+  }
+
+  // Check short names
+  for (let i = 0; i < monthNames.short.length; i++) {
+    if (lowerStr.includes(monthNames.short[i].toLowerCase())) {
+      return i + 1;
+    }
+  }
+
+  return null;
+}
+
+/**
+ * Get weekday index from Date object (Monday = 1, Sunday = 7)
+ *
+ * @param date - The date object
+ * @returns Weekday index (1-7)
+ */
+export function getWeekdayIndex(date: Date): number {
+  const day = getDay(date);
+  // Convert from Sunday=0 to Monday=1 format
+  return day === 0 ? 7 : day;
+}
+
+/**
+ * Extract day of month from a string
+ *
+ * @param dateStr - The string to extract from
+ * @returns Day of month (1-31) or null
+ */
+export function extractDayOfMonth(dateStr: string): number | null {
+  // Try parsing first
+  const date = parseFlexibleDate(dateStr);
+  if (date) {
+    return getDate(date);
+  }
+
+  // Fallback to regex
+  const match = dateStr.match(/\b([1-9]|[12]\d|3[01])\b/);
+  return match ? Number.parseInt(match[0]) : null;
+}
+
+/**
+ * Extract hour from a string
+ *
+ * @param dateStr - The string to extract from
+ * @returns Hour (0-23) or null
+ */
+export function extractHour(dateStr: string): number | null {
+  // Try parsing first
+  const date = parseFlexibleDate(dateStr);
+  if (date) {
+    return getHours(date);
+  }
+
+  // Fallback to regex for time patterns
+  const match = dateStr.match(/(\d{1,2})[:h]/);
+  return match ? Number.parseInt(match[1]) : null;
+}
+
+/**
+ * Get week number from day of year
+ *
+ * @param dayOfYear - Day of year (1-366)
+ * @returns Week number (1-53)
+ */
+export function getWeekFromDayOfYear(dayOfYear: number): number {
+  return Math.ceil(dayOfYear / 7);
+}
+
+/**
+ * Get hour of year from date
+ *
+ * @param date - The date
+ * @returns Hour of year (1-8760)
+ */
+export function getHourOfYear(date: Date): number {
+  return (getDayOfYear(date) - 1) * 24 + getHours(date) + 1;
+}
+
+/**
+ * Get the current locale from language settings
+ *
+ * @returns SupportedLocale
+ */
+export function getCurrentLocale(): SupportedLocale {
```

Comment:
There is `getCurrentLanguage` in i18nUtils

Reply:
forgot to replace it, thanks